tutor/templates/build/openedx/Dockerfile

```diff
@@ -36,20 +36,24 @@ RUN mkdir -p /openedx/edx-platform && \
 WORKDIR /openedx/edx-platform
 
 # Patch edx-platform
-# Security patches
-RUN curl https://github.com/overhangio/edx-platform/commit/a4369d300b85d94442844443a3d258a2b516955c.patch | git apply -
-# Make it possible to disable learner records globally
-# https://github.com/edx/edx-platform/pull/25182
-# https://github.com/overhangio/edx-platform/tree/overhangio/disable-learner-records-from-settings
-RUN curl https://github.com/overhangio/edx-platform/commit/bd038bab3cf02df147e754f7743e46b68b43bac8.patch | git apply -
-# Fix inconvenient pavelib warning
-# https://github.com/edx/edx-platform/pull/25771
-# https://github.com/overhangio/edx-platform/tree/overhangio/fix-paver-warning
-RUN curl https://github.com/overhangio/edx-platform/commit/bc0ab09f9945bd14aa6be1dbbf928cce58f079d2.patch | git apply -
-# Fix js upload in scorm packages by upgrading django-pipeline
-# https://github.com/edx/edx-platform/pull/25957
-# https://github.com/overhangio/edx-platform/tree/overhangio/upgrade-django-pipeline
-RUN curl https://github.com/overhangio/edx-platform/commit/1c733e3ba11249cf16358684169e6137a308e796.patch | git apply -
+RUN if [ "$EDX_PLATFORM_VERSION" = "{{ OPENEDX_COMMON_VERSION }}" ]; then \
+        # Security patches
+        curl https://github.com/overhangio/edx-platform/commit/a4369d300b85d94442844443a3d258a2b516955c.patch | git apply -; \
+        # Make it possible to disable learner records globally
+        # https://github.com/edx/edx-platform/pull/25182
+        # https://github.com/overhangio/edx-platform/tree/overhangio/disable-learner-records-from-settings
+        curl https://github.com/overhangio/edx-platform/commit/bd038bab3cf02df147e754f7743e46b68b43bac8.patch | git apply -; \
+        # Fix inconvenient pavelib warning
+        # https://github.com/edx/edx-platform/pull/25771
+        # https://github.com/overhangio/edx-platform/tree/overhangio/fix-paver-warning
+        curl https://github.com/overhangio/edx-platform/commit/bc0ab09f9945bd14aa6be1dbbf928cce58f079d2.patch | git apply -; \
+        # Fix js upload in scorm packages by upgrading django-pipeline
+        # https://github.com/edx/edx-platform/pull/25957
+        # https://github.com/overhangio/edx-platform/tree/overhangio/upgrade-django-pipeline
+        curl https://github.com/overhangio/edx-platform/commit/1c733e3ba11249cf16358684169e6137a308e796.patch | git apply -; \
+    else \
+        echo "Skip application of edx-platform repo patches since EDX_PLATFORM_VERSION does not match OPENEDX_COMMON_VERSION"; \
+    fi
```

Comment:
Thanks for adding this change @ericfab179! We're almost there. You'll need to rebase on top of the newly-released koa.3 version. But I also wanted to have your opinion on something: I think we should only exclude the patches that are already merged in koa.master, but not the custom bugfixes that are not yet present in koa.master. Thus, I would suggest the following:

```
RUN if [ "$EDX_PLATFORM_VERSION" = "{{ OPENEDX_COMMON_VERSION }}" ]; then \
        # Security patches
        curl https://github.com/overhangio/edx-platform/commit/a4369d300b85d94442844443a3d258a2b516955c.patch | git apply -; \
    else \
        echo "Skip application of edx-platform repo patches since EDX_PLATFORM_VERSION does not match OPENEDX_COMMON_VERSION"; \
    fi
# Make it possible to disable learner records globally
# https://github.com/edx/edx-platform/pull/25182
# https://github.com/overhangio/edx-platform/tree/overhangio/disable-learner-records-from-settings
RUN curl https://github.com/overhangio/edx-platform/commit/bd038bab3cf02df147e754f7743e46b68b43bac8.patch | git apply -
...
```

Basically, the idea is to allow forking from anywhere on koa.master, while still allowing custom bugfixes. But I want to make sure that this approach would work for your team with your fork. What do you think?

Reply:
@regisb thanks for the review. I was looking at this change with @ericfab179 and I was wondering if we can still have an option to bypass all these patching operations. We still have a level of customization that increases the chance of clashing with any of the patches. The issue would be those forks from koa.master not getting the custom bugfixes, but I think that if a fork needs to be maintained, these fixes could be included directly in the repo by the maintainer. Just to finish, Could it be considered a plugin for patches?

Reply:
I understand your problem @jfavellar90: you need to run a custom fork without having to worry whether patches will apply correctly -- something you currently have no control over. What I need is to ensure that most people run edx-platform with the security patches and the bugfixes: this should include people who run a fork of koa.master. Also, this should happen by default and not require the installation of a custom plugin.

To resolve both our situations, I propose an innovative solution. I just tested it and it works. Here's what it looks like:

```
FROM minimal as code
ARG EDX_PLATFORM_REPOSITORY=https://github.com/edx/edx-platform.git
ARG EDX_PLATFORM_VERSION={{ OPENEDX_COMMON_VERSION }}
RUN mkdir -p /openedx/edx-platform && \
    git clone $EDX_PLATFORM_REPOSITORY --branch $EDX_PLATFORM_VERSION --depth 1 /openedx/edx-platform
WORKDIR /openedx/edx-platform

{% if patch("openedx-dockerfile-git-patches-default") %}
# Custom edx-platform default patches
{{ patch("openedx-dockerfile-git-patches-default") }}
{% else %}
# Patch edx-platform
# Make it possible to disable learner records globally
# https://github.com/edx/edx-platform/pull/25182
# https://github.com/overhangio/edx-platform/tree/overhangio/disable-learner-records-from-settings
RUN curl https://github.com/overhangio/edx-platform/commit/bd038bab3cf02df147e754f7743e46b68b43bac8.patch | git apply -
# Fix inconvenient pavelib warning
# https://github.com/edx/edx-platform/pull/25771
# https://github.com/overhangio/edx-platform/tree/overhangio/fix-paver-warning
RUN curl https://github.com/overhangio/edx-platform/commit/bc0ab09f9945bd14aa6be1dbbf928cce58f079d2.patch | git apply -
# Fix js upload in scorm packages by upgrading django-pipeline
# https://github.com/edx/edx-platform/pull/25957
# https://github.com/overhangio/edx-platform/tree/overhangio/upgrade-django-pipeline
RUN curl https://github.com/overhangio/edx-platform/commit/1c733e3ba11249cf16358684169e6137a308e796.patch | git apply -
{% endif %}
```

What we are doing here is to create a new "openedx-dockerfile-git-patches-default" patch. If some plugin implements this patch, then we scrap the default security and bugfix patches. That way, you only have to implement a `# noop` patch in your plugin to get rid of custom patches.

Would this work for you?

Reply:
@regisb I updated the PR with your new idea which I think suites us perfectly.